lib/rucio/core/rse.py

```diff
@@ -367,7 +368,19 @@ def list_rses(filters={}, session=None):
     availability_mask2 = 7
     availability_mapping = {'availability_read': 4, 'availability_write': 2, 'availability_delete': 1}
     false_value = False  # To make pep8 checker happy ...
+
     if filters:
+        filters_to_process = len(filters)
```

Comment:
I don't understand why this `filters_to_process` stuff is needed. Could you explain?

Reply:
Following discussion with Martin implement a nicer version of this (copying the dict and popping) that doesn't alter the existing filter logic. There may be an underlying issue with the way availability filters work but don't alter that until Martin's had a further look.